lib/elementWrapper/dollarWrapper.js

```diff
@@ -17,12 +20,31 @@ class DollarWrapper extends ElementWrapper {
   constructor(attrValuePairs, _options, ...args) {
     super('CustomSelector', 'query', attrValuePairs, _options, ...args);
     this._get = async () => {
-      return await handleRelativeSearch(
-        await (this.selector.label.startsWith('//') || this.selector.label.startsWith('(')
-          ? $$xpath(this.selector.label, this._options.selectHiddenElements)
-          : $$(this.selector.label, this._options.selectHiddenElements)),
-        this.selector.args,
-      );
+      function functionSelector(argument) {
+        let { querySelector, args } = argument;
+        eval(`querySelector = ${querySelector}`);
+        const queryResult = querySelector(args);
+        if (!(queryResult instanceof Node || queryResult instanceof NodeList)) {
+          throw new Error('Query function should return a Node or NodeList');
+        }
+        return queryResult;
+      }
+
+      let element;
+      if (isFunction(attrValuePairs)) {
+        element = Element.create(
+          await runtimeHandler.findElements(functionSelector, {
+            querySelector: attrValuePairs.toString(),
+            args: _options.args,
+          }),
+          runtimeHandler,
+        );
+      } else if (this.selector.label.startsWith('//') || this.selector.label.startsWith('(')) {
```

Comment:
yes they are.. dont know the history on why we did a check only for `//` will update it

Reply:
It'll be good if we can check for valid xpath syntax (without the overhead of evaluating against the dom) using an xpath syntax checker instead of a string comparison.

Reply:
have created an issue for the same https://github.com/getgauge/taiko/issues/1595 would like that in a separate PR